scripts/dockers/pxeboot-init/bin/generate-config.py

```diff
@@ -32,17 +32,18 @@
 
 with open(target, "w") as f:
     for subgroup in ['cpu', 'storage']:
-        nodes = config['nodes'][subgroup]
-
+        nodes = config['nodes']
         f.write("#\n# %s\n#\n" % subgroup)
 
         for node in nodes:
-            f.write("%s,%s,infinite\n" % (node['management']['macaddress'].lower(), node['name']))
+            if subgroup in node['roles']:
```

Comment:
we should remove this first for loop and check if either of those roles exist write it in the config otherwise we are gonna write the same entry twice in the file for combined nodes which will cause problems